src/util.c

```diff
@@ -340,14 +335,29 @@ int ll2string(char *dst, size_t dstlen, long long svalue) {
         negative = 0;
     }
 
+    /* Converts the unsigned long long value to string*/
+    uint32_t const length = ull2string(dst, dstlen, value, negative);
+
+    /* Add sign. */
+    if (negative) dst[0] = '-';
+    return length;
+}
+
+int ull2string(char *dst, size_t dstlen, unsigned long long value, int negative) {
```

Comment:
why pass the `negative` arg to this function? instead just have the caller pass `dst+1` and fix the length.

Reply:
Negative is needed to add a +1 to lenght variable not dstlen. I tried that but it was causing errors.

Reply:
still the calling function can do all the adjustments.
* store `-` into `dst[0]`
* advance `dst` by one
* deduct one from destlen (the buffer size), 
* increment the resulting length by one.

then this function doesn't need to know anything about negative.. it's all handled by the caller.